modules/otp_pin_field/lib/src/otp_pin_field_state.dart

```diff
@@ -371,16 +371,20 @@ class OtpPinFieldState extends State<OtpPinField>
   String _getPinDisplay(int position) {
     var display = '';
     var value = pinsInputed[position];
-    switch (widget.otpPinFieldInputType) {
-      case OtpPinFieldInputType.password:
-        display = '*';
-        break;
-      case OtpPinFieldInputType.custom:
-        display = widget.otpPinInputCustom;
-        break;
-      default:
-        display = value;
-        break;
+    if( widget.obscureText == true ) {
+      display = '*'; // Use asterisks to obscure text
+    } else {
+        switch (widget.otpPinFieldInputType) {
+          case OtpPinFieldInputType.password:
+            display = '*';
+            break;
+          case OtpPinFieldInputType.custom:
+            display = widget.otpPinInputCustom;
+            break;
+          default:
+            display = value;
+            break;
+        }
     }
     return value.isNotEmpty ? display : value;
   }
```

Comment:
@TheNoumanDev this doesn't seems right, looks like this was already implemented in `otp_pin_field` module with `OtpPinFieldInputType.password` but wasn't exposed to the widget. I think the whole `OtpPinFieldInputType` enum is not exposed. and instead of duplicating the functionality, please just expose the `pinFieldType` with options `none`, `password` and `custom` and by default it should be none

Reply:
@sharjeelyunus @kmahmood74  updated PR and comment with new fields explanation and example EDL.